File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/NoticeDao.java
package mx.edu.upqroo.kristenandroid.data.database.daos;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import mx.edu.upqroo.kristenandroid.data.database.entities.Notice;

@Dao
public interface NoticeDao {
    @Insert
    long insert(Notice day);

    @Update
    void update(Notice... days);

    @Delete
    void delete(Notice... days);

    @Query("DELETE FROM notice")
    void deleteAll();

    @Query("SELECT * FROM notice WHERE noticeId = :noticeId")
    LiveData<Notice> getById(String noticeId);

    @Query("SELECT * FROM notice")
    LiveData<List<Notice>> getAll();
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/NoticeDao.kt
package mx.edu.upqroo.kristenandroid.data.database.daos

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import mx.edu.upqroo.kristenandroid.data.database.entities.Notice

@Dao
interface NoticeDao {

    @get:Query("SELECT * FROM notice")
    val all: LiveData<List<Notice>>

    @Insert
    fun insert(day: Notice): Long

    @Update
    fun update(vararg days: Notice)

    @Delete
    fun delete(vararg days: Notice)

    @Query("DELETE FROM notice")
    fun deleteAll()

    @Query("SELECT * FROM notice WHERE noticeId = :noticeId")
    fun getById(noticeId: String): LiveData<Notice>
}
